frontend/src/api/api.ts

```diff
@@ -285,15 +285,67 @@ export const apiV2 = createApi({
         data: configuration,
       }),
     }),
+    deleteIntegrationConfiguration: build.mutation<
+      void,
+      IntegrationConfiguration
+    >({
+      invalidatesTags: ['Integrations', 'Roadmaps'],
+      query: ({ roadmapId, name, id }) => ({
+        url: `roadmaps/${roadmapId}/integrations/${name}/configuration/${id}`,
+        method: 'delete',
+      }),
+    }),
+    swapIntegrationOAuthToken: build.mutation<
+      void,
+      {
+        roadmapId: number;
+        name: string;
+        swapRequest: OAuthTokenSwapRequest;
+      }
+    >({
+      invalidatesTags: ['Integrations'],
+      query: ({ roadmapId, name, swapRequest }) => ({
+        url: `/roadmaps/${roadmapId}/integrations/${name}/oauth/swaptoken`,
+        method: 'post',
+        data: swapRequest,
+      }),
+    }),
     patchIntegrationConfiguration: build.mutation<
       IntegrationConfiguration,
       IntegrationConfigurationRequest
     >({
       invalidatesTags: ['Integrations', 'Roadmaps'],
-      query: ({ roadmapId, name, id, host, consumerkey, privatekey }) => ({
+      query: ({ roadmapId, name, id, ...data }) => ({
         url: `roadmaps/${roadmapId}/integrations/${name}/configuration/${id}`,
         method: 'patch',
-        data: { host, consumerkey, privatekey },
+        data,
+      }),
+    }),
+    deleteIntegrationStatusMapping: build.mutation<
+      void,
+      { id: number; name: string; roadmapId: number; mappingId: number }
+    >({
+      invalidatesTags: ['Integrations', 'Roadmaps'],
+      query: ({ roadmapId, name, id, mappingId }) => ({
+        url: `roadmaps/${roadmapId}/integrations/${name}/configuration/${id}/statusmapping/${mappingId}`,
+        method: 'delete',
+      }),
+    }),
+    setIntegrationStatusMapping: build.mutation<
+      void,
+      {
+        id: number;
+        name: string;
+        roadmapId: number;
+        fromColumn: string;
+        toStatus: number;
```

Comment:
Would the type TaskStatus be more specific here?
```suggestion
        toStatus: TaskStatus;
```

Reply:
Yes, it would, good point